### DISCUSSION TITLE
Show time remaining on currently-playing track in playlist

### DISCUSSION THREAD

**Discussion by takaji10** [TT-ReBORN/Georgia-ReBORN#27] [Q&A] [answered]
It would be great to show the time remaining on the track that's currently playing in the playlist.

Here:
![image](https://user-images.githubusercontent.com/1534470/126347080-fa4de1be-b11f-46d1-a117-8cc200efaba4.png)

So it would show something like -2:20 if the song has been playing for a minute.

**Comment by TT-ReBORN** [comment]
I think it's enough if the playbar already has this feature now, don't want to make additional repaints on the playlist.
( Already has enough performance hits via the smooth scroll, auto-hide scrollbar and the additional show truncated tooltip).

**Reply by takaji10** [reply]
Could it be a non-default option? I personally don't use the smooth scroll and auto-hidden scrollbar.

**Comment by TT-ReBORN** [comment, marked as answer]
I have implemented and added this as an option in the Playlist -> Show time remaining on playing track
This is disabled by default. I also think I won't be implementing any new features for a while now, 
I don't want to overload and bloat this theme that has already so many features/options and cripple the performance.

**Reply by takaji10** [reply]
Thanks very much! I'll give it a test. :)

**Reply by takaji10** [reply]
Seems to be some kind of redraw/refresh error?

![image](https://user-images.githubusercontent.com/1534470/126672176-33ee209c-edf3-4e17-b831-5f78eeecb809.png)

**Reply by TT-ReBORN** [reply]
Just not a fast enough repaint, but this seems kinda weird because I set it exactly to 1 second ( 1000ms ).
Is it all the time? I have not encountered this on my PC... 

What is your CPU usage ( taskmanager foobar2000.exe ) when the option is on and when it's off?

Try to set the timer ( 1000ms ) to a lower value in Panel_Playlist.js line 5542. Experiment with different lower values.

Note: If you lower the value, the faster it repaints, ergo CPU usage will rise, so watch the taskmanager 
until you've find a good value...

**Reply by takaji10** [reply]
CPU usage with the feature enabled: ~4.3%, 243.8 MB memory
CPU usage with the feature disabled: ~1.3%, 243.6 MB memory

I'm surprised that this would create such a change to CPU usage. I've tried a few different numbers, 800, 500, still giving redraw errors... 

If it's not possible to fix the redraw issue without making the CPU usage triple, you're right in suggesting that this option doesn't get implemented. It's too bad that it's not possible to enable such a feature without such a hit to the CPU usage! :/

**Reply by TT-ReBORN** [reply]
It's already implemented :) , but it's strange that you still got redraw errors... 
I've already told you that I don't have any problems at 1000ms.

Try following, in Developer settings: Show RepaintRect areas and then enable this option.

The red rectangle should be exactly on the time where it repaints, if it's off then there is the issue.

Also try this new bugfix [here](https://github.com/TT-ReBORN/Georgia-ReBORN/commit/8258081868f69825ce9e02dc5b7f4b7a1bbfca2b).

**Reply by takaji10** [reply]
I guess your bugfix helped, because I don't see the problem anymore! And I didn't change anything. :P

Thanks for the help with making it work, I'm glad to be able to use this feature.
